enhancements/improve-kubeletconfig-default-configuration-check-enhancement.md

```diff
@@ -0,0 +1,298 @@
+  "registerNode": true,
+  "kind": "KubeletConfiguration",
+  "apiVersion": "kubelet.config.k8s.io/v1beta1"
+}
+```
+
+When we get `KubeletConfig` for each node, we will compare them, and
+only save the consistent result to a file at `/api/v1/nodes/proxy/configz`
+
+In the CaC content, we just need to check one location `/api/v1/nodes/proxy/configz`
+for `KubeletConfig` default configuration
+
+### User Stories
+
+* As an Site Reliability Engineer managing multiple clusters with many nodes, I'd like to run
+  compliance scans on my fleet in such a way that we have less compliance rule failures when
+  no reboot remediation has been applied.
+
+### API Extensions
+
+None is changed
+
+
+### Risks and Mitigations
+
+There are no risks or mitigations for this change.
+
+## Design Details
+
+### Open Questions [optional]
+
+
+1. Is `/api/v1/nodes/{NODE_NAME}/proxy/configz` a good place to check
+   default `KubeletConfig`?
+
+2. If we want to check the KubeletConfig through Kubernets API, the rule
+   needs to change from node-rule to platform rule. Since platform rules
+   does not care about node type, it will require all nodes to have the 
+   same `KubeletConfig` configuration, wondering if that is a good idea?
+
+### Test Plan
+
+It will be tested and reflected in our pre-existing CI, some `KubeletConfig` 
+checks will pass when the default `KubeletConfig` configuration is checked
+by Compliance Operator and meets the compliance requirements.
+
+
+### Upgrade / Downgrade Strategy
+
+
+Upgrade expectations:
+- No impact on existing production clusters
+
+Downgrade expectations:
+- Compliance Operator, as it is today, does not provide Downgrade options. This
+  is not expected to change.
+
+### Version Skew Strategy (TODO)
+
+How will the component handle version skew with other components?
+What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- During an upgrade, we will always have skew among components, how will this impact your work?
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI, CRI
+  or CNI may require updating that component before the kubelet.
+
+### Operational Aspects of API Extensions
+
+No API changes are introduced.
+
+#### Failure Modes
+
+- If the default `KubeletConfig` is not consistent across all nodes, some
+  `KubeletConfig` check will fail.
+
+  ex. If `makeIPTablesUtilChains": true` configuration is not consistent across all nodes,
+  we will leave out this configuration from the aggregated `KubeletConfig` result.
+  And when a compliance scan is being run, rules that checks for `makeIPTablesUtilChains`
+  will fail, because the OpenSCAP won't find `makeIPTablesUtilChains` in `/api/v1/nodes/proxy/configz`
```

Comment:
In this case, will we issue an alert or warning?

I'm not sure how this should be handled, but it feels like we should let someone know we attempted to resolve configuration differences but we couldn't.

Reply:
I think we can issue an alert for that

Reply:
Maybe a warning is better